opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -583,17 +587,28 @@ def __init__(
             OTEL_SPAN_LINK_COUNT_LIMIT,
             _DEFAULT_OTEL_SPAN_LINK_COUNT_LIMIT,
         )
+
+        # attribute count
+        global_max_attributes = self._from_env_if_absent(max_attributes, OTEL_ATTRIBUTE_COUNT_LIMIT)
+        self.max_attributes = global_max_attributes or _DEFAULT_OTEL_ATTRIBUTE_COUNT_LIMIT
+
+        self.max_span_attributes = self._from_env_if_absent(
+            max_span_attributes,
+            OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT,
+            global_max_attributes or _DEFAULT_OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT,
```

Comment:
Nit: Shouldn't `_DEFAULT_OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT` take precedence over `_DEFAULT_OTEL_ATTRIBUTE_COUNT_LIMIT`? Not an issue currently because they are both 128.

Reply:
It does. The order is as following:

user specified span limit > user specified global limit > span default > global default

(when a limit has span default, global default is never used for it)